zproject/backends.py

```diff
@@ -987,23 +987,34 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
         else:
             chosen_email = backend.strategy.request_data().get('email')
 
+        all_emails = [email['email'] for email in
+                      backend.get_all_associated_email_objects(*args, **kwargs)]
+        unverified_existing_account_emails = []
+        for email in all_emails:
+            if email not in verified_emails \
+               and common_get_active_user(email, realm, {}) is not None:
+                unverified_existing_account_emails.append(email)
+
         if not chosen_email:
             avatars = {}  # Dict[str, str]
-            existing_account_emails = []
+            verified_existing_account_emails = []
```

Comment:
renamed this for clarity. we have `existing_account_emails` for unverified accounts also. Thought it's be better to rename.